Proposed/asc055.md

```diff
@@ -0,0 +1,23 @@
+#### **Title**
+Add `Compound` to derived types
+
+#### **Authorship**
+Yang Choo <yang.choo@strateos.com>
+
+#### **Motivation**
+The usage of chemical compounds (or substances) is common throughout science. We should have some first-class representation of this type in Autoprotocol.
+
+#### **Proposal**
+Let us add a new derived type, `Compound`, which shall be represented as a `String`. Valid formats shall include the popular `SMILES` (Simplified Molecular Input Line Entry System) notation and the IUPAC `InChI` (International Chemical Identifier) notation.
+
+To differentiate between the various formats, a `notation:value` format shall be adopted, with `:` as the delimiter.
```

Comment:
I'm a proponent of this more concise format as opposed to a dictionary, as long as we don't see the need for additional keys beyond notation....none come to mind at the moment. 

I wonder if we would eventually support CAS, e.g. `CAS: 58-08-2`. CAS` and `SMILES`/`InChI`/`InChiKey` are all _identifiers_ for compounds, but somewhat different in that CAS is an "arbitrary" GUID / registry number, while the others are notations, encoding the chemical structure.

Reply:
I'm good with having a new Compound, but wondering how we will be accessing the chem database... (was there a separate discussion on this already?) We don't want to create another library for compounds in ApPy. Should we consider API dev first to access the chem database from python? ...Since APPY is public, we want to access public database?

Reply:
Are you referring to CAS? No plans to support CAS at the moment, just the molecular notations...

Reply:
@aota001 and I spoke and resolved this.

Reply:
Its totally valid to consider a dictionary. The main reason I had for not doing so was to be more conventional and consistent with our other derived types, which have a single String representation (w/ delimiters) for now. For example, `aliquot` is represented by a `ref/well` notation.
Open to change if needed in the future though.

With regards to CAS, yea, that seems like a slightly different concept to me. Focus here seems to be on the fundamental notation representation vs. some identifier. 

I think the focus of the ASC should be on the schema. It'll be up to the downstream libraries on deciding if/how to consume it and integrate with chemical databases if necessary. We could talk about the APPY-side in a separate conversation.